src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -638,7 +652,7 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
         // In case the text track on the video element is
         // different then the text track of the player
         // we need to set the correct one
-        const pkTextTrack = pkTextTracks.find(track => track.index === vidIndex);
+        const pkTextTrack = pkTextTracks.find(track => track.index === pkIndex);
```

Comment:
should be vidIndex

Reply:
i am not sure if fixed a bug / or the new flow require this change. i need to investigate why this is the the correct comparison.

Reply:
i think this part (the 'else') is logic duplication because we already trigger onTrackChange on "selectTextTrack"